File: Mushka.WebApi/ClientModels/Exhibition/ExhibitionModel.cs
﻿using System;
using System.Collections.Generic;
using Mushka.Domain.Entities;

namespace Mushka.WebApi.ClientModels.Exhibition
{
    public class ExhibitionModel
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public DateTime FromDate { get; set; }

        public DateTime ToDate { get; set; }

        public string City { get; set; }

        public decimal ParticipationCost { get; set; }

        public PaymentMethod ParticipationCostMethod { get; set; }

        public decimal Profit { get; set; }

        public string Notes { get; set; }

        public IEnumerable<ExhibitionProductModel> Products { get; set; }
    }
}

File: frontend/src/app/shared/models/exhibition.model.ts
import { ExhibitionProduct } from './exhibition-product.model';
import { PaymentMethod } from '../enums/payment-method.enum';

export class Exhibition {
  id: string;
  name: string;
  fromDate: string;
  toDate: string;
  city: string;
  participationCost: number;
  participationCostMethod: PaymentMethod;
  notes: string;
  profit: number;
  products: ExhibitionProduct[];

  accommodationСost: number;
  accommodationСostMethod: PaymentMethod;
  fareCost: number;
  fareCostMethod: PaymentMethod;

  constructor(data: any) {
    Object.assign(this, data);
  }
}
